### ISSUE TITLE
Issue with drawing the circuit (Latex output)

### ISSUE THREAD

**Issue by ericrsong** [Qiskit/qiskit-terra#1818]
### Informations

- **Qiskit version**: 
qiskit                   0.7.2
qiskit-aer               0.1.1
qiskit-aqua              0.4.1
qiskit-chemistry         0.4.2
qiskit-terra             0.7.0
- **Python version**: 3.7
- **Operating system**: windows 10 x64

### What is the current behavior?
When plotting the circuit, the barrier is not in the correct place when choosing the latex output. 

Output of `circuit.draw(output='latex')`:
![fig1](https://user-images.githubusercontent.com/47463894/52884500-27f9ec80-313c-11e9-9d10-fd885a9ab9ab.JPG)

Reversing the bit removes one Hadamard gate on qa_2. 
Output of `circuit.draw(output='latex', reverse_bits=True)`:
![fig2](https://user-images.githubusercontent.com/47463894/52884511-30eabe00-313c-11e9-91a6-bcecc9b49de8.JPG)
Same issue appears on the documentation: [https://qiskit.org/documentation/terra/visualizing_a_quantum_circuit.html](url)
### Steps to reproduce the problem

```import matplotlib.pyplot as plt
from qiskit import QuantumCircuit, ClassicalRegister, QuantumRegister
from qiskit import BasicAer, execute
backend = BasicAer.get_backend('qasm_simulator')


##Draw a new circuit with barriers and more registers

q_a = QuantumRegister(3, name='qa')
q_b = QuantumRegister(5, name='qb')
c_a = ClassicalRegister(3)
c_b = ClassicalRegister(5)

circuit = QuantumCircuit(q_a, q_b, c_a, c_b)

circuit.x(q_a[1])
circuit.x(q_b[1])
circuit.x(q_b[2])
circuit.x(q_b[4])
circuit.barrier()
circuit.h(q_a)
circuit.barrier(q_a)
circuit.h(q_b)
circuit.cswap(q_b[0], q_b[1], q_b[2])
circuit.cswap(q_b[2], q_b[3], q_b[4])
circuit.cswap(q_b[3], q_b[4], q_b[0])
circuit.barrier(q_b)
circuit.measure(q_a, c_a)
circuit.measure(q_b, c_b);



##Draw the circuit
circuit.draw(output='latex')


##Draw the circuit with reversed bit order
circuit.draw(output='latex', reverse_bits=True)
```

### What is the expected behavior?
Output of `circuit.draw(output='mpl')`:
![figure_1](https://user-images.githubusercontent.com/47463894/52884972-a7d48680-313d-11e9-8d31-ca3dc0331c78.png)

### Suggested solutions
Please check the code for plotting latex image or the latex package being used.

**Comment by mtreinish** [comment]
This looks like a duplicate of #1724 which was already fixed on master. Can you test that #1727 fixes this for you? (it does for me locally)

**Comment by ericrsong** [comment]
Sorry, yes. The Hadamard gate appears now. The barrier still seems to be not in the exact right place. But that's not a big issue.

![circuit](https://user-images.githubusercontent.com/47463894/53045374-728ca900-345b-11e9-998a-ceb481f48344.JPG)

Btw, there's an incompatibility between versions.
![incompatible](https://user-images.githubusercontent.com/47463894/53045433-99e37600-345b-11e9-9f6c-10fb2d524106.JPG)

**Comment by mtreinish** [comment]
Yeah, the barrier placement can be pretty finicky. It's a known issue (and there is limits to what is doable there) because the latex package requires manually offsetting the barrier horizontally so we have to try to account for every spacing in the python code when generating the latex. 

As for that error, it'll go away when we bump the major version on the `qiskit` package. The versioning on the `qiskit` package doesn't allow for dev versions, there's probably a better way to handle that. But that's a separate discussion to have on the https://github.com/Qiskit/qiskit repo.

I'm going to close this issue as already fixed by #1727 feel free to open new issues for the barrier overlap and/or the dev versions with the metapackage.
